s3_resource/src/s3_operations.cpp

```diff
@@ -327,13 +327,11 @@ namespace irods_s3 {
                     }
             }
 
-            // if number of threads was not successfully set above
+            // If number of threads was not successfully set above.
             if (number_of_threads == 0) {
-
                 const int single_buff_sz = irods::get_advanced_setting<const int>(irods::KW_CFG_MAX_SIZE_FOR_SINGLE_BUFFER) * 1024 * 1024;
-                number_of_threads = requested_number_of_threads;
 
-                if (data_size > single_buff_sz && oprType != REPLICATE_DEST && oprType != COPY_DEST) {
```

Comment:
Did we reach a conclusion about the `oprType` conditions?

Reply:
> Did we reach a conclusion about the `oprType` conditions?

No, unfortunately those oprType conditions were added while migrating away from S3FS.  It is just one commit now and I think I have lost the history of why they were added.  (Should have had a better comment.)

Tests are passing though and removing COPY_DEST did fix #2204.